src/Http/Headers/src/SetCookieHeaderValue.cs

```diff
@@ -141,61 +140,75 @@ public override string ToString()
                 length += SeparatorToken.Length + HttpOnlyToken.Length;
             }
 
-            var sb = new InplaceStringBuilder(length);
+            return string.Create(length, (this, maxAge), (span, tuple) =>
+            {
+                var (headerValue, maxAgeValue) = tuple;
 
-            sb.Append(_name);
-            sb.Append(EqualsToken);
-            sb.Append(_value);
+                Append(ref span, headerValue._name);
+                Append(ref span, EqualsToken);
+                Append(ref span, headerValue._value);
 
-            if (expires != null)
-            {
-                AppendSegment(ref sb, ExpiresToken, expires);
-            }
+                if (headerValue.Expires is DateTimeOffset expiresValue)
```

Comment:
How does this handle null?

Reply:
`DateTimeOffset` is a struct so it evaluates to `false`.